java/server/src/org/openqa/selenium/grid/distributor/local/LocalDistributor.java

```diff
@@ -206,6 +205,29 @@ private LocalDistributor add(Node node, NodeStatus status) {
   }
 
   @Override
+  public boolean drain(NodeId nodeId) {
+    Lock writeLock = lock.writeLock();
+    writeLock.lock();
+    try {
+      Optional<Host> host = hosts.stream().filter(node -> nodeId.equals(node.getId())).findFirst();
+      if(host.isPresent())
+      {
+        org.openqa.selenium.grid.data.Status status = host.get().drainHost();
+
+        return status == org.openqa.selenium.grid.data.Status.DRAINING;
```

Comment:
Comment: We get away with this because enums are interned, but it's safer to do `DRAINING.equals(status)`. There's no risk of an NPE, and if we ever change the type of `Status` we're safe.